libkbfs/folder_branch_ops.go

```diff
@@ -1176,6 +1176,16 @@ func (fbo *folderBranchOps) SetInitialHeadFromServer(
 		return fmt.Errorf("MD with revision=%d not initialized", md.Revision())
 	}
 
+	// Return early if the head is already set.  This avoids taking
```

Comment:
OK. Although I suspect that the code would be simplified if we e.g. made this function do the MD fetch instead of the caller (since it's already doing an MD fetch anyway to get the latest merged revision)